### DISCUSSION TITLE
Eclipse install issue, can you lend a hand?

### DISCUSSION THREAD

**Discussion by anonimo82** [nathaneltitane/dextop#22] [General]
Hello everybody. I know, it's not a Dextop fault, but I don't know where else to ask for help.
Please, can you help me? I can't run Eclipse, see attached screenshot.
![Screenshot_20230503-174616_bVNC Free](https://user-images.githubusercontent.com/38261009/235968647-4945d6bc-0925-4d8c-8b08-829dd3d96877.jpg)
And this is what happens from the command line:
![Screenshot_20230503-174745_Termux](https://user-images.githubusercontent.com/38261009/235969029-c342ed24-6df9-47cf-9083-b192cdfdad86.jpg)
Thanks for any help.

Regards =)

**Comment by nathaneltitane** [comment]
Hello @anonimo82,

I am not aware as to what eclipse actually is or does. From what I can tell, the binary or executable you are trying to run is either inextistant or does not have the right permissions to be run.

I assume from your prompt (in green) that you are running it within the Termux part of the Dextop container.

Does your eclipse app/package need an actual distribution to run?

**Comment by anonimo82** [comment]
I think it does, but is this important?
I mean, is a full distro really required?

**Reply by nathaneltitane** [reply]
> I think it does, but is this important? I mean, is a full distro really required?

Is your 'eclipse' the java compiler?
If so did you try to install it from the termux repositories irst?

If not , then yes, you dod need a full distribution. Do you understand the difference in between Termux and a full distro?

**Comment by anonimo82** [comment]
Eclipse is a popular IDE for various programming languages.
I downloaded a binary archive, extracted and tried to execute it, but then, the error.
With a full distro, no problem.

I didn't understand the difference with a full distro.

**Reply by nathaneltitane** [reply]
> Eclipse is a popular IDE for various programming languages. I downloaded a binary archive, extracted and tried to execute it, but then, the error. With a full distro, no problem.
> 
> I didn't understand the difference with a full distro.

The major difference lies within the librabies used on host systems and what Eclipse, in this case, was compiled against.

Most of these differences are linked to the use of libc - a major library used for compiling applications for runtime usage - see this: https://wiki.termux.com/wiki/Differences_from_Linux

I would suggest running Dextop using the Ubuntu container that is built and unpacked on setup and running Eclipse from in there - might aslo be better if Eclipse has a package already available from repositories as you are almost certain it will work with close to no fuss.

**Comment by anonimo82** [comment]
Yes, I'm using container with "container-session -o"
Is it that what you're talking about?

**Reply by nathaneltitane** [reply]
> Yes, I'm using container with "container-session -o" Is it that what you're talking about?

That's the normal Termux-side container - you need to launch with `container-session -u termux` your prompt should be fuschia.

Please review the readme:

![image](https://user-images.githubusercontent.com/5674757/236459824-02a2efd9-cabf-4695-866e-527cae479b90.png)

**Comment by anonimo82** [comment]
Hey, works!
Thanks a lot!
